sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -239,10 +270,23 @@ static az_result connecting(az_event_policy* me, az_event event)
   switch (event.type)
   {
     case AZ_HFSM_EVENT_ENTRY:
+    {
+      _az_RETURN_IF_FAILED(az_platform_clock_msec(&this_policy->_internal.connect_start_time_msec));
+      break;
+    }
     case AZ_HFSM_EVENT_EXIT:
-      // No-op.
+    {
+      int64_t current_time_msec = 0;
```

Comment:
Nit: extract an action function here. Given that the file is growing fairly big, consider moving all action functions in a separate compilation unit. That way the state machine code is easier to review and compare with the design.

Reply:
Please document this in the PUML source (something on the lines of
```
enter/ stopwatch_reset()
exit / elapsed_time_msec = stopwatch_gettime()
```